virttest/migration.py

```diff
@@ -578,6 +579,45 @@ def set_migratepostcopy(self, vm_name, uri=None):
             raise exceptions.TestError("Unalbe to find event {}"
                                        .format(exp_str))
 
+    def control_migrate_speed(self, vm_name, to_speed=1, opts=""):
+        """
+        Set migration speed to control the migration duration
+
+        :param vm_name: vm name
+        :param to_speed: the speed value in Mbps to be set for migration
+        :param opts: the options for virsh command
+        """
+
+        def _set_speed(extra_option):
+            """
+            Inner function to set migration speed
+
+            :param extra_option: str, it might include '--postcopy' or not
+            :raises: exceptions.TestError when not support
+            """
+            if ('--postcopy' in extra_option and
+                    not libvirt_version.version_compare(5, 0, 0)):
+                raise exceptions.TestError("libvirt version should be larger "
+                                           "than or equal to '5.0.0' when "
+                                           "setting postcopy migration speed")
```

Comment:
Is 'speed set' a 'must have' step? If no, I think we'd better log an error/return directly rather than raising TestError to stop testing. What do you think?

Reply:
updated